MaterialLib/SolidModels/ViscoElasticBurgers-impl.h

```diff
@@ -0,0 +1,332 @@
+        _state.KM * eps_i_trace *
+            MaterialLib::SolidModels::Invariants<KelvinVectorSize>::identity2;
+
+    // Calculate dGdE for time step
+    Eigen::Matrix<double, KelvinVectorSize * 3, KelvinVectorSize,
+                  Eigen::RowMajor> const dGdE = CaldGdEBurgers();
+
+    // If the Newton loop didn't run, the linear solver will not be initialized.
+    // This happens usually for the first iteration of the first timestep.
+    if (counter == 0)
+        linear_solver.compute(K_loc);
+
+    // Consistent tangent from local Newton iteration of material
+    // functionals.
+    // Only the upper left block is relevant for the global tangent.
+    auto dzdE = linear_solver.solve(-1.0 * dGdE)
+                    .template block<KelvinVectorSize, KelvinVectorSize>(0, 0);
+
+    auto const& P_sph = MaterialLib::SolidModels::Invariants<
+        KelvinVectorSize>::spherical_projection;
+    C = _state.GM * dzdE * P_dev + 3. * _state.KM * P_sph;
+}
+
+template <int DisplacementDim>
+void ViscoElasticBurgers<DisplacementDim>::UpdateBurgersProperties(
+    double const t,
+    ProcessLib::SpatialPosition const& x,
+    double s_eff,
+    MaterialStateVariables& _state)
+{
+    _state.GM = _mp.GM0(t, x)[0];
+    _state.KM = _mp.KM0(t, x)[0];
+    _state.GK = _mp.GK0(t, x)[0] * std::exp(_mp.mK(t, x)[0] * s_eff);
+    _state.etaK = _mp.etaK0(t, x)[0] * std::exp(_mp.mvK(t, x)[0] * s_eff);
+    _state.etaM = _mp.etaM0(t, x)[0] * std::exp(_mp.mvM(t, x)[0] * s_eff);
+}
+
+template <int DisplacementDim>
+void ViscoElasticBurgers<DisplacementDim>::CalResidualBurgers(
+    const double dt,
+    const typename ViscoElasticBurgers<DisplacementDim>::KelvinVector&
+        strain_curr,
+    const typename ViscoElasticBurgers<DisplacementDim>::KelvinVector&
+        stress_curr,
+    typename ViscoElasticBurgers<DisplacementDim>::KelvinVector&
+        strain_Kel_curr,
+    const typename ViscoElasticBurgers<DisplacementDim>::KelvinVector&
+        strain_Kel_t,
+    typename ViscoElasticBurgers<DisplacementDim>::KelvinVector&
+        strain_Max_curr,
+    const typename ViscoElasticBurgers<DisplacementDim>::KelvinVector&
+        strain_Max_t,
+    typename ViscoElasticBurgers<DisplacementDim>::ResidualVector& res,
+    MaterialStateVariables const& _state)
+{
+    // calculate stress residual
+    res.template block<KelvinVectorSize, 1>(0, 0).noalias() =
+        stress_curr - 2. * (strain_curr - strain_Kel_curr - strain_Max_curr);
+
+    // calculate Kelvin strain residual
+    res.template block<KelvinVectorSize, 1>(KelvinVectorSize, 0).noalias() =
+        1. / dt * (strain_Kel_curr - strain_Kel_t) -
+        1. / (2. * _state.etaK) *
+            (_state.GM * stress_curr - 2. * _state.GK * strain_Kel_curr);
+
+    // calculate Maxwell strain residual
+    res.template block<KelvinVectorSize, 1>(2 * KelvinVectorSize, 0).noalias() =
+        1. / dt * (strain_Max_curr - strain_Max_t) -
+        _state.GM / (2. * _state.etaM) * stress_curr;
```

Comment:
I don't see the point here. Neither the mathematical expression reads better (to me) nor is there any performance difference; Could you explain the suggestion please?